http.h

```diff
@@ -40,7 +40,9 @@ struct http_uri {
 	int port;
 };
 
+struct connection;
 struct http_get {
+	int is_https;
 	struct http_uri uri;
 	struct http_uri *proxy;
```

Comment:
Since your uri parser function does not support ports i.e. `[:port]` and most often proxy run on different ports than the default 80 or 443 how does proxy support even work?